### PR TITLE
Replace Enumerations::TableCell by ViewComponent

### PR DESCRIPTION
**Pull request by cbliard**
https://community.openproject.org/wp/47009

This is part of a larger effort to replace Cells by ViewComponents.

### CODE REVIEW

File: app/components/enumerations/row_component.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#++
+
+module Enumerations
+  class RowComponent < ::RowComponent
+    def enumeration
+      row
+    end
+
+    def name
+      link_to h(enumeration.name), edit_enumeration_path(enumeration)
```
Comment (ulferts) [inline app/components/enumerations/row_component.rb]: I believe the `h` is unnecessary as the link_to will already escape everything by now.

File: app/components/table_component.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  # # required by the sort helper
+
+  # delegate :controller_name, to: :controller
+
+  # delegate :action_name, to: :controller
+
+  # def options
+  #   super
+  # end
```
Comment (ulferts) [inline app/components/table_component.rb]: leftovers?

**Review by ulferts — CHANGES_REQUESTED** [review]
Just tiny remarks

File: app/components/enumerations/row_component.rb (hunk 1 above)
Comment (cbliard) [reply app/components/enumerations/row_component.rb]: You're right
```
[1] pry(main)> include ActionView::Helpers
=> Object
[2] pry(main)> link_to('Hello, <b>World</>!', 'https://example.com')
=> "<a href=\"https://example.com\">Hello, &lt;b&gt;World&lt;/&gt;!</a>"
```
I'll fix it.

File: app/components/table_component.rb (hunk 2 above)
Comment (cbliard) [reply app/components/table_component.rb]: Yep. I'll remove them.

**Pull request merged**